### PR TITLE
Support gitlab (modified version of nakatanakatana )

### PR DESCRIPTION
**Pull request by masamotod**
I fixed based on @nakatanakatana's PR.
https://github.com/haya14busa/reviewdog/pull/121

* Fix commented points in #121 
* Merge current master branch
  * `-reporter=gitlab-mr-review` option, etc.

### CODE REVIEW

File: cmd/reviewdog/main.go [hunk 1]
```diff
@@ -303,6 +324,58 @@ func githubBaseURL() (*url.URL, error) {
...
+	}
+
+	gitlabservice, err = reviewdog.NewGitLabMergeReqest(client, g.Owner, g.Repo, g.PullRequest, g.SHA)
+	if err != nil {
+		return nil, isPR, err
+	}
+	return gitlabservice, isPR, nil
+}
+
+func gitlabClient(_ context.Context, token string) (*gitlab.Client, error) {
```
Comment (haya14busa) [inline cmd/reviewdog/main.go]: Can you remove unused context arg?

File: gitlab.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		if _, ok := g.postedcs[path][pos]; !ok {
+			g.postedcs[path][pos] = make([]string, 0)
+		}
+		g.postedcs[path][pos] = append(g.postedcs[path][pos], body)
+	}
+	return nil
+}
+
+// Diff returns a diff of MergeRequest. It runs `git diff` locally instead of
+// diff_url of GitLab Pull Request because diff of diff_url is not suited for
```
Comment (haya14busa) [inline gitlab.go]: [super nit] s/Pull Request/Merge Request/

File: gitlab_test.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+	}
+	comment := &Comment{
+		CheckResult: &CheckResult{
+			Path: "diff.go",
+			Lnum: 22,
+		},
+		LnumDiff: 11,
+		Body:     "[reviewdog] test",
+	}
+	// https://gitlab.com/nakatanakatana/reviewdog/merge_requests/1
```
Comment (haya14busa) [inline gitlab_test.go]: I don't have a permission to this repository.
Can you remove the test which actually sends a request to GitLab?

File: cmd/reviewdog/main.go (hunk 1 above)
Comment (masamotod) [reply cmd/reviewdog/main.go]: Removed ✨

File: gitlab.go (hunk 2 above)
Comment (masamotod) [reply gitlab.go]: Fixed 🐶

File: gitlab_test.go (hunk 3 above)
Comment (masamotod) [reply gitlab_test.go]: Removed 3 tests.
How about to give repo info as environment variables?
(I may send another pull request)

Comment (haya14busa) [reply gitlab_test.go]: Sounds good.
But the tests assume the content of the merge-request, so I'll first create hard-coded tests.

I can merge p-r to make it env var too.
Also, it's totally optional but if you want to run tests by yourself with the repository I'll create, I'll give you a permission.

**Comment by haya14busa** [comment]
Thank you for your work, @masamotod and @nakatanakatana :+1: :100:

**Comment by masamotod** [comment]
thank youuuuuuuuuu

**Pull request merged**